src/network.zig

```diff
@@ -991,30 +994,27 @@ pub const Protocols = struct {
 		};
 
 		fn receive(conn: *Connection, reader: *utils.BinaryReader) !void {
-			switch(try reader.readInt(u8)) {
-				type_gamemode => {
+			const updateType = reader.readEnum(UpdateType) catch |err| {
+				std.log.err("Error {s} in generic protocol message: '{s}'", .{@errorName(err), reader.remaining});
+				return error.Invalid;
+			};
```

Comment:
Using `try` is enough for network interfaces. This should only happen when connecting with the wrong version and the player doesn't need to know these details in that case. (also trying to print binary data with {s} is also incorrect, since that would print it as a string)

Reply:
Btw why do we have all those different interfaces instead of just two, one for "important" and one for "unimportant" stuff?

Reply:
What do you mean? We'd need the same number of protocols either way.